Modules/Devices/src/Devices.Application/Identities/Commands/CancelDeletionProcessAsSupport/Handler.cs

```diff
@@ -21,19 +21,20 @@ public Handler(IIdentitiesRepository identitiesRepository, IEventBus eventBus)
 
     public async Task<CancelDeletionAsSupportResponse> Handle(CancelDeletionAsSupportCommand request, CancellationToken cancellationToken)
     {
+        var identity = await _identitiesRepository.FindByAddress(request.Address, cancellationToken) ?? throw new NotFoundException(nameof(Identity));
+
         var deletionProcessIdResult = IdentityDeletionProcessId.Create(request.DeletionProcessId);
 
         if (deletionProcessIdResult.IsFailure)
             throw new DomainException(deletionProcessIdResult.Error);
 
         var deletionProcessId = deletionProcessIdResult.Value;
 
-        var identity = await _identitiesRepository.FindWithDeletionProcess(deletionProcessId, cancellationToken, true) ?? throw new NotFoundException(nameof(Identity));
-        var deletionProcess = identity.CancelDeletionProcessAsSupport();
+        var deletionProcess = identity.CancelDeletionProcessAsSupport(deletionProcessId);
 
         await _identitiesRepository.Update(identity, cancellationToken);
 
-        _eventBus.Publish(new IdentityDeletionProcessStartedIntegrationEvent(identity.Address, deletionProcess.Id));
+        _eventBus.Publish(new IdentityDeletionProcessStatusChangedIntegrationEvent(identity.Address, deletionProcess.Id));
```

Comment:
I just realized something: I think we have to trigger 2 more integration events:
* TierOfIdentityChanged
* IdentityStatusChanged

Can you make sure they are also triggered in case of the cancellation as owner? There not even the IdentityDeletionProcessStatusChangedIntegrationEvent is triggered. :(

Reply:
There is no `IdentityStatusChanged` integration event. I will assume you are talking about the `IdentityDeletionProcessStatusChangedIntegrationEvent`.

Reply:
Oh, wow. I thought we had that. But okay, then ignore the second one, and only trigger TierOfIdentityChanged.